src/convertToObject.js

```diff
@@ -12,7 +12,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+
+  const arr = sourceString.split(';').map(
+    el => el.split(':')
+  ).filter(a => a.length > 1);
```

Comment:
you can avoid filter method by simply checking if elements exist before adding to an object.